version: '3.7'

services:
  auth:
    image: dnikolaidis/disastergram-auth:nginx
    build:
      context: .
      dockerfile: Dockerfile-nginx
    deploy:
      replicas: 2
      restart_policy:
        condition: on-failure
    environment:
      - NGINX_WORKER_PROCESSES=4
      - FLASK_APP_INSTANCE=/app/instance
      - FLASK_APP_PERSISTENT=/app/persistent
      - FLASK_APP_MIGRATIONS=/app/persistent/migrations
      - FLASK_APP=/app/auth
    volumes:
      - auth-persistent:/app/persistent
    ports:
    - "5000:80"
    networks:
      - auth-net
    depends_on:
      - auth-db

  # https://hub.docker.com/_/postgres
  auth-db:
    image: postgres:11
    restart: always
    environment:
      POSTGRES_PASSWORD: disastergram
    volumes:
      - auth-data:/var/lib/postgresql/
    networks:
      - auth-net

networks:
  auth-net:

volumes:
  auth-data:
  auth-persistent: